sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -415,6 +415,11 @@ case class Literal (value: Any, dataType: DataType) extends LeafExpression {
   Literal.validateLiteralValue(value, dataType)
 
   override def foldable: Boolean = true
+
+  override def contextIndependentFoldable: Boolean = {
+    !DataTypeUtils.matchesPattern(dataType, dt => dt.isInstanceOf[UserDefinedType[_]])
```

Comment:
How can a literal be context dependent? It's not an operation.

Reply:
Looking at `Literal.validateLiteralValue`:
```
case udt: UserDefinedType[_] => doValidate(v, udt.sqlType)
```

So the stored value is the same as UDT's underlying SQL type. I don't think it requires any context

Reply:
For UDT, users can customize how the data is `serialize` and `deserialize`. For example, the serialized result can be concated with current date or current user.
Let's be conservative here.

Reply:
The catalyst expression evaluation does not involve any UDT `serialize`/`deserialize`, right?